src/main/java/org/opensearch/neuralsearch/transport/MLPredictAction.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.transport;
+
+import org.opensearch.action.ActionType;
+import org.opensearch.common.io.stream.Writeable;
+
+public class MLPredictAction extends ActionType<MLPredictActionResponse> {
```

Comment:
Mentioned this below, but why do we need this transport Action? Why cant we just use ML-commons transport actions directly?

Reply:
we cannot use the Transport action present in the MLCommons as MLClient is supposed to abstract that.